lib/IRGen/GenDecl.cpp

```diff
@@ -926,16 +926,22 @@ void IRGenModule::emitVTableStubs() {
                                     llvm::GlobalValue::InternalLinkage,
                                     "_swift_dead_method_stub");
       stub->setAttributes(constructInitialAttributes());
+      // FIXME: should we be setting the visibility and DLL storage as well?
+
       Module.getFunctionList().push_back(stub);
       stub->setCallingConv(DefaultCC);
       auto *entry = llvm::BasicBlock::Create(getLLVMContext(), "entry", stub);
       auto *errorFunc = getDeletedMethodErrorFn();
       llvm::CallInst::Create(errorFunc, ArrayRef<llvm::Value *>(), "", entry);
       new llvm::UnreachableInst(getLLVMContext(), entry);
     }
+
     // For each eliminated method symbol create an alias to the stub.
-    llvm::GlobalAlias::create(llvm::GlobalValue::ExternalLinkage, F.getName(),
-                              stub);
+    auto *alias = llvm::GlobalAlias::create(llvm::GlobalValue::ExternalLinkage,
+                                            F.getName(), stub);
+    if (TargetInfo.OutputObjectFormat == llvm::Triple::COFF)
+      alias->setDLLStorageClass(llvm::GlobalValue::DLLExportStorageClass);
+    // FIXME: should we be setting the visibility as well?
```

Comment:
LLVM IR does not track whether visibility has been set or not.  Calling the constructor is guaranteed to make the visibility default.

Accordingly, features like -fvisibility=hidden are implemented in frontends by changing the visibility computation.